### ISSUE TITLE
Add new option for one-shot code evaluation

### ISSUE THREAD

**Issue by jvican** [tpolecat/tut#134]
It would be interesting to have one-shot evaluation blocks. Currently, code snippets are evaluated expression by expression and, because of pecularities of how the REPL works, it may interest some users to evaluate everything at once which should produce the same result. This option could also be useful to get rid of printing of intermediate results that may be unnecessary.
  
If this sounds good, I would be happy to give it a try.

**Comment by tpolecat** [comment]
You mean per-shed? Are you thinking something that works like `:paste` mode in the REPL?

**Comment by jvican** [comment]
Yeah, exactly, the same behaviour as `:paste`.

**Comment by tpolecat** [comment]
Yep, there's an open issue about that and I was just talking to someone else about it on the gitter channel. I'll take a look after work and see if I can sketch out a strategy … I tried before and got hung up somewhere but don't remember where.

**Comment by jvican** [comment]
It's great that's ongoing work. I've just realised that there's also an option in the REPL `-Yrepl-class-based`that would solve the reason why I needed this to be evaluated at once. If this is already open in another issue, feel free to close it.

**Comment by tpolecat** [comment]
Oh I don't know what that option does. I'll take a look.

**Comment by jvican** [comment]
It basically changes the way the REPL works internally. Instead of wrapping every expression in an object, it does so in classes that are dynamically loaded and nested to avoid scoping issues.

**Comment by ssimeonov** [comment]
👍 on adding `:paste` support. I'd be cautious about relying on `-Yrepl-class-based` as well as anything that makes the environment where `tut` executes different from that of the people following the generated documentation.

**Comment by rossabaker** [comment]
:+1: I'm doing a presentation in tut tonight (which is The Way To Do Presentations) and just had to add a `tut:invisible` to get my typeclass instances into implicit scope.
